src/main/java/bio/terra/service/snapshot/SnapshotTableDao.java

```diff
@@ -93,42 +111,59 @@ private void createColumns(UUID tableId, Collection<Column> columns) {
   }
 
   public List<SnapshotTable> retrieveTables(UUID parentId) {
-    MapSqlParameterSource params = new MapSqlParameterSource().addValue("parent_id", parentId);
-    return jdbcTemplate.query(
+    MapSqlParameterSource params = new MapSqlParameterSource().addValue("snapshot_id", parentId);
+    Map<UUID, SnapshotTable> tableMap = new TreeMap<>();
+    jdbcTemplate.query(
         sqlSelectTable,
         params,
         (rs, rowNum) -> {
+          UUID tableId = rs.getObject("table_id", UUID.class);
           SnapshotTable table =
-              new SnapshotTable()
-                  .id(rs.getObject("id", UUID.class))
-                  .name(rs.getString("name"))
-                  .rowCount(rs.getLong("row_count"));
-          List<Column> columns = retrieveColumns(table);
-          table.columns(columns);
+              tableMap.computeIfAbsent(
+                  tableId,
+                  (id) -> {
+                    try {
+                      return new SnapshotTable()
+                          .id(id)
+                          .name(rs.getString("table_name"))
+                          .rowCount(rs.getLong("table_row_count"))
+                          .columns(new ArrayList<>())
+                          .primaryKey(new ArrayList<>());
```

Comment:
Since this is a type we wrote (vs one openapi generates), you could add a new constructor that does some of this work internally.
```suggestion
                      return new SnapshotTable(id, rs.getString("table_name"), rs.getLong("table_row_count"));
```

Reply:
Eh, I like the fluent approach.  We talked offline about eventually moving this to a builder pattern